### ISSUE TITLE
important bug - Stack says it installed binaries, but in fact, it doesnt

### ISSUE THREAD

**Issue by rvion** [commercialhaskell/stack#1042]
ping @snoyberg @borsboom 

Stack says it installed binaries, but in fact, it doesnt

---------

while `stack upgrade --git`,

```
[...]
[1 of 2] Compiling Paths_stack      ( .stack-work/dist/x86_64-osx/Cabal-1.22.4.0/build/autogen/Paths_stack.hs, .stack-work/dist/x86_64-osx/Cabal-1.22.4.0/build/stack/stack-tmp/Paths_stack.o )
[2 of 2] Compiling Main             ( src/main/Main.hs, .stack-work/dist/x86_64-osx/Cabal-1.22.4.0/build/stack/stack-tmp/Main.o )
Linking .stack-work/dist/x86_64-osx/Cabal-1.22.4.0/build/stack/stack ...
Completed all 3 actions.
.: canonicalizePath: does not exist (No such file or directory)
```

even though it says `Completed all 3 actions`, it didn't

probable cause:  `.: canonicalizePath: does not exist (No such file or directory)`

`stack --version` says `Version 0.1.5.0, Git revision 58f6ebc78ba391ef987cf6e502e597dbcd6672db X86_64` instead of `c759ccf7708b9746b90333bed61684484a1e90c3`

is it related to https://github.com/commercialhaskell/stack/pull/1019 ?

**Comment by rvion** [comment]
now, I directly get `.: canonicalizePath: does not exist (No such file or directory)`
```
stack upgrade --git
.: canonicalizePath: does not exist (No such file or directory)
```

**Comment by rvion** [comment]
just downloaded the released 1.5 version and it seems to be working

```
stack --version
Version 0.1.5.0, Git revision 1893348f11301684e64567c93b49f9e0ceb602f9 (2154 commits) X86_64
```
